android/src/main/java/com/applovin/reactnative/AppLovinMAXAdView.java

```diff
@@ -83,57 +169,94 @@ public void maybeAttachAdView(final String placement, final String customData, f
             return;
         }
 
-        // Run in next run loop when view is laid out and `getWidth()` / `getHeight()` has appropriate values
-        post( new Runnable()
+        if ( TextUtils.isEmpty( adUnitId ) )
+        {
+            AppLovinMAXModule.e( "Attempting to attach MaxAdView without Ad Unit ID" );
+            return;
+        }
+
+        if ( adFormat == null )
+        {
+            AppLovinMAXModule.e( "Attempting to attach MaxAdView without ad format" );
+            return;
+        }
+
+        // Re-assign in case of race condition
+        final String adUnitId = this.adUnitId;
+        final MaxAdFormat adFormat = this.adFormat;
+
+        // Run after 0.25 sec delay to allow all properties to set
+        postDelayed( new Runnable()
         {
             @Override
             public void run()
             {
-                // If ad unit id and format has been set - create and attach AdView
-                if ( !TextUtils.isEmpty( adUnitId ) && adFormat != null )
+                if ( adView != null )
                 {
-                    adView = new MaxAdView( adUnitId, adFormat, AppLovinMAXModule.getInstance().getSdk(), currentActivity );
-                    adView.setListener( AppLovinMAXModule.getInstance() );
-                    adView.setRevenueListener( AppLovinMAXModule.getInstance() );
+                    AppLovinMAXModule.e( "Attempting to re-attach with existing MaxAdView: " + adView );
+                    return;
+                }
 
-                    if ( placement != null )
-                    {
-                        adView.setPlacement( placement );
-                    }
+                AppLovinMAXModule.d( "Attaching MaxAdView..." );
 
-                    if ( customData != null )
-                    {
-                        adView.setCustomData( customData );
-                    }
+                adView = new MaxAdView( adUnitId, adFormat, AppLovinMAXModule.getInstance().getSdk(), currentActivity );
+                adView.setListener( AppLovinMAXModule.getInstance() );
+                adView.setRevenueListener( AppLovinMAXModule.getInstance() );
 
-                    if ( adaptiveBannerEnabledStr != null )
-                    {
-                        adView.setExtraParameter( "adaptive_banner", adaptiveBannerEnabledStr );
-                    }
+                if ( placement != null )
+                {
+                    adView.setPlacement( placement );
+                }
 
-                    // Set this extra parameter to work around a SDK bug that ignores calls to stopAutoRefresh()
-                    adView.setExtraParameter( "allow_pause_auto_refresh_immediately", "true" );
+                if ( customData != null )
+                {
+                    adView.setCustomData( customData );
+                }
 
-                    if ( autoRefreshEnabled != null )
+                if ( adaptiveBannerEnabled != null )
```

Comment:
If adaptive banners are enabled / disabled - where do we factor in the adaptive size?

https://dash.applovin.com/documentation/mediation/react-native/getting-started/banners#adaptive-banners

Reply:
Javascript defines the adaptive size for the view.    Android follows the size.

```
  const sizeForAdFormat = () => {
    if (props.adFormat === AdFormat.BANNER) {
      let width = AppLovinMAX.isTablet() ? 728 : 320;
      let height;

      if (props.adaptiveBannerEnabled) {
        height = AppLovinMAX.getAdaptiveBannerHeightForWidth(-1);
      } else {
        height = AppLovinMAX.isTablet() ? 90 : 50;
      }

      return { width: width, height: height }
    } else {
      return { width: 300, height: 250 }
    }
  };

```